sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -402,6 +404,28 @@ public Read withRowFilter(RowFilter filter) {
       return withRowFilter(StaticValueProvider.of(filter));
     }
 
+    /**
+     * Returns a new {@link BigtableIO.Read} that will break up read requests into smaller batches.
+     * This function will switch the base BigtableIO.Reader class to using the SegmentReader. If
+     * null is passed, this behavior will be disabled and the stream reader will be used.
+     *
+     * <p>Does not modify this object.
+     *
+     * <p>When we have a builder, we initialize the value. When they call the method then we
+     * override the value
+     */
+    @Experimental(Kind.SOURCE_SINK)
+    public Read withMaxBufferElementCount(@Nullable Integer maxBufferElementCount) {
```

Comment:
Can we add an end-to-end integration test for this (can be done in a separate PR) ?

Reply:
I think that would be a great idea, what file contains the integration tests for BigtableIO? I should be able to get it done before the end of this half.